.github/workflows/tests.yml

```diff
@@ -0,0 +1,36 @@
+name: Tests
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+jobs:
+  tests_linux:
+    runs-on: ${{ matrix.os }}
+    needs: []
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9, "3.10"]
+        os: ["ubuntu-latest", "macOS-latest", "windows-2019"]
```

Comment:
I think what you did here is reasonable.  It tests on every Python version on each supported operating system, for a total of 12 tests.

I'll just mention that in the prototype template, I did it a little bit differently.  I have it only test on all Python versions on Linux and test on a single Python version on the other platforms.  This saves a bit of CI time, yet still allows us to detect anything that would break on a different version of Python or on a different platform.  I suspect that the full matrix is generally not necessary, but it's a reasonable thing to do if you think it's worth it.  We can always try it here and determine later what works best for us across the board.

https://github.com/qiskit-community/quantum-prototype-template/blob/c909c853e6c05626177d7708fc220a85c1c4f2a2/.github/workflows/test_latest_versions.yml#L20-L27